fast64_internal/oot/c_writer/oot_scene_table_c.py

```diff
@@ -1,75 +1,123 @@
-
-	if newFileData != fileData:
-		writeFile(os.path.join(exportPath, 'src/code/z_scene_table.c'), newFileData)
+def getSceneTable(exportPath):
+	'''Read and remove unwanted stuff from ``scene_table.h``'''
+	dataList = []
+	fileHeader = ""
+	debugLine = ""
+
+	# read the scene table
+	try:
+		with open(os.path.join(exportPath, 'include/tables/scene_table.h')) as fileData:
+			# keep the relevant data and do some formatting
+			for i, line in enumerate(fileData):
+				if not line.startswith("// "):
+					if not (line.startswith("/**") or line.startswith(" *")):
+						dataList.append(line[(line.find("(") + 1):].rstrip(")\n").replace(" ", "").split(','))
+					else: fileHeader += line
+				elif not line.startswith("// Added scenes"): debugLine = line
+	except: raise PluginError("ERROR: Can't find scene_table.h!")
+
+	# return the parsed data, the header comment and the comment mentionning debug scenes
+	return dataList, fileHeader, debugLine
+
+def getSceneIndex():
+	'''Returns the index (int) of the chosen scene, returns None if ``Custom`` is chose'''
+	i = 0
+	sceneID = bpy.context.scene.ootSceneOption
+
+	if sceneID == "Custom": return None
+	for elem in ootEnumSceneID:
+		if elem[0] == sceneID: return i - 1
+		i += 1
+
+	raise PluginError("ERROR: Scene Index not found!")
+
+def getSceneParams(scene, exportInfo):
+	'''Returns the parameters that needs to be set in ``DEFINE_SCENE()``'''
+	sceneIndex = getSceneIndex()
+	sceneName = sceneTitle = sceneID = sceneUnk10 = sceneUnk12 = None
+
+	# if the index is None then this is a custom scene
+	if sceneIndex == None and scene != None:
+		sceneName = scene.name.lower() + "_scene"
+		sceneTitle = "none"
+		sceneID = "SCENE_" + (scene.name.upper() if scene is not None else exportInfo.name.upper())
+		sceneUnk10 = sceneUnk12 = 0
+
+	return sceneName, sceneTitle, sceneID, sceneUnk10, sceneUnk12, sceneIndex
+
+def sceneTableToC(data, header, debugLine):
+	'''Converts the Scene Table to C code'''
+	# start the data with the header comment explaining the format of the file
+	fileData = header
+	# used to add the "// Added scenes" comment after existing scenes
+	isEndOfExistingScenes = False
+	# used to add the "// Debug-only scenes" comment after non-debug scenes
+	isEndOfNonDebugScenes = False
+
+	# add the actual lines with the same formatting
+	for i in range(len(data)):
+		# add a comment to show when it's new scenes
+		if isEndOfExistingScenes:
+			fileData += "// Added scenes\n"
+			isEndOfExistingScenes = False
+		
+		# adds the "// Debug-only scenes" comment after SCENE_GANON_TOU
+		if isEndOfNonDebugScenes:
+			fileData += debugLine
+			isEndOfNonDebugScenes = False
+
+		fileData += f"/* 0x{i:02X} */ DEFINE_SCENE("
+		for j in range(len(data[i])):
+			fileData += f"{data[i][j]}"
+			if j < 5: fileData += ", "
+			if data[i][0] == "testroom_scene": isEndOfExistingScenes = True
+			if data[i][0] == "ganon_tou_scene": isEndOfNonDebugScenes = True
```

Comment:
I think this can be outside the `for j` loop
```suggestion
		if data[i][0] == "testroom_scene": isEndOfExistingScenes = True
		if data[i][0] == "ganon_tou_scene": isEndOfNonDebugScenes = True
```

Reply:
I really don't like how I named ``isEndOfExistingScenes`` and ``isEndOfNonDebugScenes`` there btw, I just couldn't find a better descriptive name